aidb/engine/approx_aggregate_engine.py

```diff
@@ -161,12 +165,12 @@ async def execute_inference_services(self, query: Query, sample_df: pd.DataFrame
           new_query
       )
 
-      input_df = await conn.run_sync(lambda conn: pd.read_sql(text(query_add_filter_key.sql_str), conn))
+      input_df = await conn.run_sync(lambda conn: pd.read_sql_query(text(query_add_filter_key.sql_str), conn))
       await bound_service.infer(input_df)
       inference_services_executed.add(bound_service.service.name)
 
 
-  async def get_results_for_each_blob(self, query: Query, sample_df: pd.DataFrame, conn) -> pd.DataFrame:
+  async def get_results_for_each_blob(self, query: Query, sample_df: pd.DataFrame) -> pd.DataFrame:

```

Comment:
Restart the engine within the function; there's no need to pass 'conn'.

Reply:
Why do we need this change? Is it possible to separate this out into a separate PR?

Reply:
In MySQL database, after running inference (there is another engine.begin() in 'infer()' function), the database will lose connection. We need to restart the engine, which is implemented inside 'get_results_for_each_blob'. So there's no need to pass 'conn'.